[BE][doc] add memory_profiler to README (#606)
pytorch/torchtitan: docs/memory_profiler.md

## Enable Memory Profiling

Launch training job with the following command (or alternatively set configs in toml files)
```
CONFIG_FILE="./train_configs/debug_model.toml" ./run_llama_train.sh --profiling.enable_memory_snapshot --profiling.save_memory_snapshot_folder output_folder
```
* `--profiling.enable_memory_snapshot`: enable memory snapshot

> * `--profiling.save_memory_snapshot_folder`: dump memory snapshots in to output folder, default to be `memory_snapshot`.

Reviewer: The default here should be `./outputs/memory_snapshot`

Reviewer: I mean the default value for this field is `memory_snapshot`, but you should mention it would be under the output folder, which is controlled by `job.dump_folder`.

	+ If in case of OOMs. output folder is `memory_snapshot/iteration_x_exit`.

> 	+ If regularly according to `profile_freq`. output folder is `memory_snapshot/iteration_x`.

Reviewer 2: explain how people should open the pickle file? like go to browser, drag, and check. we should quote https://pytorch.org/blog/understanding-gpu-memory-1/

## Enable Memory Profiling

Launch training job with the following command (or alternatively set configs in toml files)
```

> CONFIG_FILE="./train_configs/debug_model.toml" ./run_llama_train.sh --profiling.enable_memory_snapshot --profiling.save_memory_snapshot_folder output_folder

Reviewer: ```suggestion
CONFIG_FILE="./train_configs/debug_model.toml" ./run_llama_train.sh --profiling.enable_memory_snapshot --profiling.save_memory_snapshot_folder memory_snapshot
```

```

> * `--profiling.enable_memory_snapshot`: enable memory snapshot

Reviewer: ```suggestion
* `--profiling.enable_memory_snapshot`: to enable memory profiling
```

> * `--profiling.save_memory_snapshot_folder`: dump memory snapshots in to output folder, default under your output folder to be `./outputs/memory_snapshot`.

Reviewer: ```suggestion
* `--profiling.save_memory_snapshot_folder`: configures the folder which memory snapshots are dumped into (`./outputs/memory_snapshot/` by default)
```

> 	+ If in case of OOMs. output folder is `memory_snapshot/iteration_x_exit`.

Reviewer 3: The first "." should be replace with ",".

> 	+ If in case of OOMs, output folder is `memory_snapshot/iteration_x_exit`.

Reviewer: ```suggestion
	+ In case of OOMs, the snapshots will be in `./outputs/memory_snapshot/iteration_x_exit`.
```

> 	+ If regularly according to `profile_freq`. output folder is `memory_snapshot/iteration_x`.

Reviewer 3: ditto.

> 	+ If regularly according to `profile_freq`, output folder is `memory_snapshot/iteration_x`.

Reviewer: ```suggestion
	+ Regular snapshots (taken every `profiling.profile_freq` iterations) will be in `memory_snapshot/iteration_x`.
```

> Once you have dumped the memory profiler, you will find the saved pickle files in your output folder.

Reviewer: Let's remove this sentence, as it is redundant.

> To visualize the snapshot file, you can utilize the `memory_viz` tool, by either dragging and dropping the snapshot into your browser or generating its HTML file, following the [tutorial](https://pytorch.org/blog/understanding-gpu-memory-1/).

Reviewer: ```suggestion
To visualize a snapshot file, you can drag and drop it to <https://pytorch.org/memory_viz>. To learn more details on memory profiling, please visit this [tutorial](https://pytorch.org/blog/understanding-gpu-memory-1/).
```
